drivers/spi/spi_handlers.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <spi.h>
+#include <syscall_handler.h>
+
+#ifndef CONFIG_SPI_LEGACY_API
+_SYSCALL_HANDLER(spi_transceive, config_p, tx_bufs, tx_count, rx_bufs,
+		 rx_count)
+{
+	struct spi_config *config = (struct spi_config *)config_p;
+
+	_SYSCALL_MEMORY_READ(config, sizeof(*config));
+	_SYSCALL_OBJ(config->dev, K_OBJ_DRIVER_SPI);
+
+	if (tx_count) {
+		_SYSCALL_MEMORY_ARRAY_READ(tx_bufs, tx_count,
+					   sizeof(struct spi_buf));
+	}
+
+	if (rx_count) {
+		_SYSCALL_MEMORY_ARRAY_WRITE(rx_bufs, rx_count,
+					    sizeof(struct spi_buf));
```

Comment:
Ditto: need to validate the spi_buf struct itself, but also the memory it points to.